### PR TITLE
Add FTPFile tests and fix Javadoc typos

### PR DESCRIPTION
**Pull request by jkbkupczyk**
- Fixed some typos in Javadoc and adjusted documentation to code
- Replaced HTML snippets with Javadoc counterparts
- Moved serialization-related methods under the "serialization comment"
- Adds FTPFile tests

FYI @kinow @garydgregory

### CODE REVIEW

**Comment by garydgregory** [comment]
Hi @jkbkupczyk 

Please don't move methods in PRs, it makes your PR noisier, and takes longer to review. The methods were sorted in AB order, let's keep it that way, nice and simple, easier to find code, especially in PRs ;-)

**Comment by jkbkupczyk** [comment]
@garydgregory done

Thought it looked kinda odd to have the `read` method before setters. But it makes sense to have them sorted :)

File: src/main/java/org/apache/commons/net/ftp/FTPFile.java [hunk 1]
```diff
@@ -330,13 +330,14 @@ public void setName(final String name) {
...
      * constants) to the file.
      *
-     * @param access     The access group (one of the <code> _ACCESS </code> constants)
-     * @param permission The access permission (one of the <code> _PERMISSION </code> constants)
-     * @param value      True if permission is allowed, false if not.
+     * @param access     The access group (one of the {@code _ACCESS} constants)
+     * @param permission The access permission (one of the {@code _PERMISSION} constants)
+     * @param value      {@code true} if permission is allowed, {@code false} if not.
      * @throws ArrayIndexOutOfBoundsException if either of the parameters is out of range
+     * @throws NullPointerException if you're trying to set permission of an invalid file
```
Comment (garydgregory) [inline src/main/java/org/apache/commons/net/ftp/FTPFile.java:340]: How can an NPE happen here?

File: src/main/java/org/apache/commons/net/ftp/FTPFile.java [hunk 2]
```diff
@@ -466,6 +467,9 @@ public String toFormattedString(final String timezone) {
 
     /**
      * Gets a string representation of the FTPFile information.
+     * The returned output will be an FTP server raw listing.
+     *
+     * @see #getRawListing()
      *
```
Comment (garydgregory) [inline src/main/java/org/apache/commons/net/ftp/FTPFile.java:473]: No blank line between tags.

File: src/main/java/org/apache/commons/net/ftp/FTPFile.java [hunk 3]
```diff
@@ -466,6 +467,9 @@ public String toFormattedString(final String timezone) {
 
     /**
      * Gets a string representation of the FTPFile information.
+     * The returned output will be an FTP server raw listing.
```
Comment (garydgregory) [inline src/main/java/org/apache/commons/net/ftp/FTPFile.java:470]: The whole Javadoc can just say "Delegates to ..." and use a link tag to refer to the method.

File: src/main/java/org/apache/commons/net/ftp/FTPFile.java (hunk 1 above)
Comment (jkbkupczyk) [reply src/main/java/org/apache/commons/net/ftp/FTPFile.java:340]: [FTPListParseEngine](https://github.com/apache/commons-net/blob/master/src/main/java/org/apache/commons/net/ftp/FTPListParseEngine.java) will use `FTPFile(String)` constructor that sets `permissions` to `null`;

File: src/main/java/org/apache/commons/net/ftp/FTPFile.java (hunk 3 above)
Comment (jkbkupczyk) [reply src/main/java/org/apache/commons/net/ftp/FTPFile.java]: done

File: src/main/java/org/apache/commons/net/ftp/FTPFile.java (hunk 2 above)
Comment (jkbkupczyk) [reply src/main/java/org/apache/commons/net/ftp/FTPFile.java]: done, I also removed the blank line in `toFormattedString` Javadoc

**Pull request merged**